hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -448,6 +448,7 @@ private List<MergeOnReadInputSplit> getInputSplits(
     final String mergeType = this.conf.getString(FlinkOptions.MERGE_TYPE);
     return readPartitions.stream()
         .map(relPartitionPath -> getFileSlices(fsView, relPartitionPath, maxQueryInstantTime, skipBaseFiles)
+            .sorted((Comparator.comparing(FileSlice::getBaseInstantTime)))
```

Comment:
We already have a sort in line 467?

Reply:
hi, latestCommit is the latest instant, and we need to sort the baseInstantTime of each FileSlice

Reply:
@empcl : Since we are sorting by getLastCommit() at the end of this processing step, wouldn't it defeat the purpose of ordering by base Instant time ? 

Can you also write a simple unit-test to verify the expected order ?

Reply:
> need to sort the baseInstantTime of each FileSlice

They have the same semantics inside one file group?

Reply:
> @empcl : Since we are sorting by getLastCommit() at the end of this processing step, wouldn't it defeat the purpose of ordering by base Instant time ?
> 
> Can you also write a simple unit-test to verify the expected order ?

Okay, let me improve it